File: cordova/scripts/setup_gradle_properties.js
const fs = require("fs");
const path = require("path");

const PROPS = {
  "android.useAndroidX": "true",
  "android.enableJetifier": "true"
};

module.exports = function(ctx) {
  // Make sure android platform is part of build
  if (!ctx.opts.platforms.includes("android")) return;

  const platformRoot = path.join(ctx.opts.projectRoot, "platforms/android");
  const gradlePropertiesFile = path.join(platformRoot, "gradle.properties");

  const lines = fs
    .readFileSync(gradlePropertiesFile)
    .toString()
    .split("\n");

  for (let key in PROPS) {
    let value = PROPS[key];
    let replaced = false;

    // Try and replace existing
    for (let i in lines) {
      const line = lines[i];
      if (line.indexOf(key) === 0) {
        lines[i] = `${key}=${value}`;
        replaced = true;
      }
    }

    // Add if missing
    if (!replaced) {
      lines.push(`${key}=${value}`);
    }
  }

  fs.writeFileSync(gradlePropertiesFile, lines.join("\n") + "\n");
};

File: cordova/scripts/setup_gradle_properties.py
#!/usr/bin/env python

import os
import sys

target_props = {
  "android.useAndroidX": "true",
  "android.enableJetifier": "true"
};

cordova_path = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..'))
gradle_prop_path = os.path.abspath(os.path.join(cordova_path, 'platforms/android/gradle.properties'))

fd = open(gradle_prop_path, "r")
lines = fd.read().split('\n')
fd.close()

for k, v in target_props.items():
  replaced = False

  for i in range(len(lines)): 
    line = lines[i];
    if line.find(k) == 0:
      lines[i] = k + "=" + v
      replaced = True
      break

  if not replaced:
    lines.append(k + "=" + v)


fd = open(gradle_prop_path, "w")
fd.write("\n".join(filter(len, lines)))
fd.close(); 
